content/client-lib-development-guide/features.textile

```diff
@@ -870,6 +871,7 @@ h4. Message
 ** @(TM2g)@ @name@ string
 ** @(TM2d)@ @data@ string, buffer or JSON-encodable object or array
 ** @(TM2e)@ @encoding@ string
+** @(TM2i)@ @extras@ JSON-encodeable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-encodeable objects or JSON-encodeable arrays. For context, the @extras@ field is planned for use with push notifications as a payload in the @push@ key
```

Comment:
I think that we should say explicitly that the `extras` field is provided to contain message metadata and/or ancillary payloads in support of specific functionality, eg push. Each of these supported extensions is documented separately; for 0.9 the only supported extension is push, via the `extras.push` member. The processing of any other members is undefined.

As written here it sounds like you can freely put anything in there, which is wrong.